### PR TITLE
refactor(ui_tests): move ui_tests to test_utils crate to reduce development time

### PR DESCRIPTION
**Pull request by jagan-jaya**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

UI tests only depend on selenium library and few common utils in tests, it doesn't need any other things present in router. This PR move all the ui_tests related changes to test_utils crate so that development time can be reduced while adding unit test. 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Currently, writing a ui_tests takes around ~30secs for each change done in *_ui.rs or selenium.rs file. this can be reduced to 3secs by moving all the ui_tests related changes to test_utils crate as it has very less dependency with other modules.

## How did you test it?

ran UI tests manually in pipeline. it worked as expected

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed submitted code

### CODE REVIEW

File: crates/test_utils/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#![allow(
```
Comment (jarnura) [inline crates/test_utils/tests/mod.rs:1]: use tests.rs as file name, avoid using mod.rs conventions as we don't flow that.

File: crates/test_utils/tests/main.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#![allow(
```
Comment (jarnura) [inline crates/test_utils/tests/main.rs:1]: if this is just a util library rename the file to lib.rs

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — DISMISSED** [review]

**Review by jarnura — APPROVED** [review]

File: crates/test_utils/tests/connectors.rs
Comment (SanchithHegde) [inline crates/test_utils/tests/connectors.rs:1]: Do we need this file?

File: crates/test_utils/tests/integration_tests.rs
Comment (SanchithHegde) [inline crates/test_utils/tests/integration_tests.rs:1]: Do we need this file as well?

File: crates/test_utils/tests/lib.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#![allow(
+    clippy::expect_used,
+    clippy::panic,
+    clippy::unwrap_in_result,
+    clippy::unwrap_used
+)]
```
Comment (SanchithHegde) [inline crates/test_utils/tests/lib.rs:6]: If you're adding it here, can you remove this annotation from the files corresponding to each of the submodules?

File: crates/test_utils/tests/lib.rs
Comment (SanchithHegde) [inline crates/test_utils/tests/lib.rs:1]: Can we have the directory structure similar to how we've done for the connectors (have a directory named `connectors` within the `tests` directory, have a `main.rs` instead, etc.)?
https://github.com/juspay/hyperswitch/tree/f32fdec290a2f303887550d8db1ae2a3c065bafe/crates/router/tests/connectors

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by jarnura — APPROVED** [review]

**Pull request merged**